metadata/src/main/java/org/apache/kafka/metadata/authorizer/StandardAuthorizer.java

```diff
@@ -129,23 +167,37 @@ public synchronized void loadSnapshot(Map<Uuid, StandardAcl> acls) {
     public List<AuthorizationResult> authorize(
             AuthorizableRequestContext requestContext,
             List<Action> actions) {
-        StandardAuthorizerData curData = data;
         List<AuthorizationResult> results = new ArrayList<>(actions.size());
-        for (Action action: actions) {
-            AuthorizationResult result = curData.authorize(requestContext, action);
-            results.add(result);
+        lock.readLock().lock();
+        try {
+            for (Action action : actions) {
+                AuthorizationResult result = data.authorize(requestContext, action);
```

Comment:
Even in the lock, we should still load `data` into a local variable to avoid loading the volatile multiple times